sdk/storage/azblob/internal/exported/blob_batch.go

```diff
@@ -0,0 +1,269 @@
+
+	batchSubRequest.WriteString(fmt.Sprintf("%s %s %s%s", req.Raw().Method, blobPath, HttpVersion, HttpNewline))
+
+	for k, v := range req.Raw().Header {
+		if strings.EqualFold(k, shared.HeaderXmsVersion) {
+			continue
+		}
+		if len(v) > 0 {
+			batchSubRequest.WriteString(fmt.Sprintf("%v: %v%v", k, v[0], HttpNewline))
+		}
+	}
+
+	batchSubRequest.WriteString(HttpNewline)
+	return []byte(batchSubRequest.String())
+}
+
+// CreateBatchRequest creates a new batch request using the sub-requests present in the BlobBatchBuilder.
+//
+// Example of a sub-request in the batch request body:
+//
+//	--batch_357de4f7-6d0b-4e02-8cd2-6361411a9525
+//	Content-Type: application/http
+//	Content-Transfer-Encoding: binary
+//	Content-ID: 0
+//
+//	DELETE /container0/blob0 HTTP/1.1
+//	x-ms-date: Thu, 14 Jun 2018 16:46:54 GMT
+//	Authorization: SharedKey account:G4jjBXA7LI/RnWKIOQ8i9xH4p76pAQ+4Fs4R1VxasaE=
+//	Content-Length: 0
+func CreateBatchRequest(bb *BlobBatchBuilder) ([]byte, string, error) {
+	batchID, err := createBatchID()
+	if err != nil {
+		return nil, "", err
+	}
+
+	// Create a new multipart buffer
+	reqBody := &bytes.Buffer{}
+	writer := multipart.NewWriter(reqBody)
+
+	// Set the boundary
+	err = writer.SetBoundary(batchID)
+	if err != nil {
+		return nil, "", err
+	}
+
+	partHeaders := make(textproto.MIMEHeader)
+	partHeaders["Content-Type"] = []string{"application/http"}
+	partHeaders["Content-Transfer-Encoding"] = []string{"binary"}
+	var partWriter io.Writer
+
+	for i, req := range bb.SubRequests {
+		if bb.AuthPolicy != nil {
+			resp, err := bb.AuthPolicy.Do(req)
+			if err != nil && resp != nil {
+				if log.Should(EventSubmitBatch) {
+					log.Writef(EventSubmitBatch, "failed to authorize sub-request for %v.\nError: %v\nResponse status: %v", req.Raw().URL.Path, err.Error(), resp.Status)
+				}
+			}
+		}
+
+		partHeaders["Content-ID"] = []string{fmt.Sprintf("%v", i)}
+		partWriter, err = writer.CreatePart(partHeaders)
+		if err != nil {
+			return nil, "", err
+		}
+
+		_, err = partWriter.Write(buildSubRequest(req))
+		if err != nil {
+			return nil, "", err
+		}
+	}
+
+	// Close the multipart writer
+	err = writer.Close()
+	if err != nil {
+		return nil, "", err
+	}
+
+	return reqBody.Bytes(), batchID, nil
+}
+
+// UpdateSubRequestHeaders updates the sub-request headers.
+// Removes x-ms-version header.
+func UpdateSubRequestHeaders(req *policy.Request) {
+	// remove x-ms-version header from the request header
+	for k := range req.Raw().Header {
+		if strings.EqualFold(k, shared.HeaderXmsVersion) {
+			delete(req.Raw().Header, k)
+		}
+	}
+}
+
+// BlobBatchSubResponse contains the response for the individual sub-requests.
+type BlobBatchSubResponse struct {
+	ContentID     *int
+	ContainerName *string
+	BlobName      *string
+	RequestID     *string
+	Version       *string
+	Error         error // nil error indicates that the batch sub-request operation is successful
+}
+
+func getResponseBoundary(contentType *string) (string, error) {
```

Comment:
See `mime.ParseMediaType()` to get the response boundary.

Reply:
Done